### ISSUE TITLE
[netrunner and maui]: since we ship firefox in both, we need to seperate this?

### ISSUE THREAD

**Issue by star-buck** [netrunner-common/netrunner-settings-firefox#1]
or is there any other method in setting different bookmarks for maui website and netrunner?

if not, lets clone both from here into:
- https://github.com/maui-common
- https://github.com/netrunner-core (because the corebased variants like owncloud could yet have other bookmarks).

**Comment by llelectronics** [comment]
Agreed. We need to split it then if we want to have different bookmarks.

**Comment by star-buck** [comment]
eventually also have different addons, like netrunner core has less, but for now only bookmarks.
maybe there is a better way to have firefox-common and ship the bookmarks as netrunner-core-firefox-bookmarks?

**Comment by llelectronics** [comment]
This is something we need to check. For what I know we have to ship the whole Mozilla profile. Though it should be possible to ship a global bookmarks config.

**Comment by NuLogicSystems** [comment]
There is, you can use a distrobution.ini file like we do with Manjaro:
https://github.com/manjaro/manjaro-browser-settings/tree/master/firefox

This would end up under /usr/lib/firefox/distribution/

**Comment by NuLogicSystems** [comment]
Any other changes like adding in the plugins could go under  /usr/lib/firefox as well.
For example adblock-plus could be placed here: 
usr/lib/firefox/browser/extensions/{d10d0bf8-f5b5-c8b4-a8b2-2b9879e08c5d}.xpi

This would also speed up loading since the ~.mozilla directory wouldn't need to be copied from skel.
The same could also be adopted for Thunderbird.

**Comment by llelectronics** [comment]
Yeah also discussed that with Rohan aswell. Most problematic is the anttoolbar plugin that isn't provided as xpi file. That breaks every now and then.

**Comment by NuLogicSystems** [comment]
Yea, Ant Video Downloader also seems to get updated often.

**Comment by llelectronics** [comment]
@AJSlye The distribution.ini file seems only to work half way. It adds those bookmarks but all other ones (the default ones) stay. 
Setting the startpage also seems not to work. It always loads its own about:startpage thing. This is the section I used for my tests:

```
[LocalizablePreferences]
browser.startup.homepage="https://www.google.com"
browser.startup.homepage_override.mstone=ignore
```

**Comment by NuLogicSystems** [comment]
Yes, the distribution.ini is limited in what it can do, you can use it to add bookmarks and to change some settings. I think what your looking for is bookmarks.initialized.pref: under [Global]:
https://wiki.mozilla.org/Distribution_INI_File

However, for some of what your want to do, a custom profile.js file, etc. might be needed.

**Comment by NuLogicSystems** [comment]
This guy has same great content on customization of firefox:
He used to be a consultant but now works for mozilla.
https://mike.kaply.com/

**Comment by llelectronics** [comment]
Yeah saw that already. Not everything though still possible with the described posts. But I found a way to create a autoconfig.js that defines a config file to load settings. But that seems to have limits aswell. 
Removing predefined bookmarks for example, however setting up the UI rudimentary should be possible. 
I guess no one else does customize Firefox like we do therefore it takes some time to figure that all out and to create the perfect config

**Comment by NuLogicSystems** [comment]
Yea, that's what I figured as well, most seem to be happy with just adding some bookmarks and changing the homepage.
